### PR TITLE
Remove deploy_staking_contract_by_system

### PR DESCRIPTION
**Pull request by cna0**
This pull-request addresses the issue #3678.
Deployment of the staking contract should not be built-in.
`deploy_staking_contract_by_system` was meant for testing purposes, however, I couldn't find any tests relying on it.

### CODE REVIEW

File: apps/aehyperchains/src/aehc_consensus_hyperchains.erl [hunk 1]
```diff
@@ -817,54 +805,6 @@ get_hc_activation_criteria() ->
...
-                    Value = aect_call:return_type(Call),
-                    aec_consensus:config_assertion_failed("Failed to deploy staking contract", " Error: ~p Ret: ~p\n", [What, Value])
-            end,
-            cleanup_system_owner(OldA, Trees2);
-        {error, What} ->
-            aec_consensus:config_assertion_failed("Unable to deploy staking contract by system", " Error: ~p\n", [What])
-    end.
 
 %% Temporarily grants a lot of funds for the system account
 prepare_system_owner(Deployer, Trees) ->
```
Comment (cytadela8) [inline apps/aehyperchains/src/aehc_consensus_hyperchains.erl:810]: Please remove also prepare_system_owner and cleanup_system_owner. These should be unused now.

**Review by cytadela8 — COMMENTED** [review]
Logic of state_pre_transform_key_node_consensus_switch could (line 363) be also simplified

**Review by cytadela8 — APPROVED** [review]
Great!

**Comment by cytadela8** [comment]
Rebase please

**Comment by cna0** [comment]
@cytadela8 Done.

I think we should merge #3647 first, and then I will come back updating this PR.

File: apps/aecore/src/aec_consensus_common_tests.erl [hunk 2]
```diff
@@ -186,7 +186,22 @@ genesis_difficulty() -> 0.
 %% -------------------------------------------------------------------
 %% Keyblock creation
 new_unmined_key_node(PrevNode, PrevKeyNode, Height, Miner, Beneficiary, Protocol, InfoField, TreesIn) ->
-    aec_consensus_bitcoin_ng:new_unmined_key_node(PrevNode, PrevKeyNode, Height, Miner, Beneficiary, Protocol, InfoField, TreesIn).
+    FakeBlockHash = <<0:?BLOCK_HEADER_HASH_BYTES/unit:8>>,
```
Comment (cytadela8) [inline apps/aecore/src/aec_consensus_common_tests.erl:189]: Looks like you got in rebase commits that shouldn't be there.

**Review by cytadela8 — CHANGES_REQUESTED** [review]
Fix the rebase

**Comment by cytadela8** [comment]
Make sure when rebasing that your local hyperchain-master is up-to-date.

**Comment by cna0** [comment]
@cytadela8 Just a side note: rebasing from the target branch is most of the time unnecessary when there's no conflicts.

**Review by cytadela8 — APPROVED** [review]

**Comment by seanhinde** [comment]
Superseded by new HC design

**Pull request closed without merging**